pyshader/_generator_bc.py

```diff
@@ -188,7 +227,9 @@ def co_entrypoint(self, name, shader_type, execution_modes):
 
     def co_func_end(self):
         if self._current_branch is not self._root_branch:
-            raise RuntimeError("Function ends with unresolved sub-branches!")
+            raise RuntimeError(
```

Comment:
Maybe create a helper function called `error` which takes the message, and optionally some variables and raises the `ShaderError`?

Reply:
I considered that, but I don't like the parrern of calling a function to raise an error, in part because it ads an extra layer in the stack trace.